integration_test/cases/backoff_test.exs

```diff
@@ -16,12 +28,19 @@ defmodule BackoffTest do
         {:ok, :state}
       end]
     {:ok, agent} = A.start_link(stack)
+    agent
+  end
 
-    opts = [agent: agent, parent: self(), backoff_min: 10]
+  defp execute_test_backoff_after_failed(agent, opts, test_hibernate? \\ false) do
     {:ok, _} = P.start_link(opts)
     assert_receive {:error, conn}
     assert_receive {:hi, ^conn}
 
+    if test_hibernate? and Mix.env() != :sojourn do
+      assert {:current_function, {:erlang, :hibernate, 3}} ==
+        Process.info(conn, :current_function)
```

Comment:
This may have a race condition as we can't guarantee the process has hibernated because of the async nature of processes. However we can leave this until its observed.